crates/ty_python_semantic/src/types.rs

```diff
@@ -2164,6 +2164,25 @@ impl<'db> Type<'db> {
                 true
             }
 
+            (
+                Type::Callable(_) | Type::DataclassDecorator(_) | Type::DataclassTransformer(_),
+                Type::NominalInstance(instance),
+            )
+            | (
+                Type::NominalInstance(instance),
+                Type::Callable(_) | Type::DataclassDecorator(_) | Type::DataclassTransformer(_),
+            ) if instance.class.is_final(db) => {
+                let member = self.member_lookup_with_policy(
+                    db,
+                    Name::new_static("__call__"),
+                    MemberLookupPolicy::NO_INSTANCE_FALLBACK,
+                );
+                match member.symbol {
+                    Symbol::Type(ty, _) => !ty.is_assignable_to(db, CallableType::unknown(db)),
```

Comment:
```suggestion
                    // TODO: ideally this would check disjointness of the `__call__` signature and the callable signature
                    Symbol::Type(ty, _) => !ty.is_assignable_to(db, CallableType::unknown(db)),
```

Reply:
I'm removing the bad comment in https://github.com/astral-sh/ruff/pull/18384